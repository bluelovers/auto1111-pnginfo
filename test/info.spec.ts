//@noUnusedParameters:false
/// <reference types="jest" />
/// <reference types="node" />
/// <reference types="expect" />

import { basename, extname, join } from 'path';
import { readFileSync } from 'fs';
import parseFromImageBuffer from '../src/index';

beforeAll(async () =>
{

});

describe(basename(__filename, extname(__filename)), () =>
{
	let buf: Buffer;
	beforeEach(() => {
		buf = readFileSync(join(__dirname, 'tmp46c8k1iv.png'))
		;
	});

	test.skip(`dummy`, () => {});

	test(`PNGINFO`, async () =>
	{
		expect(parseFromImageBuffer(buf)).toMatchSnapshot();

	});

	test(`PNGINFO:cast_to_snake`, async () =>
	{
		expect(parseFromImageBuffer(buf, true)).toMatchSnapshot();
	});

})
