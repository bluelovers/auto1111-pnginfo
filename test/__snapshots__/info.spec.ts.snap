// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`info.spec PNGINFO 1`] = `
{
  "metadata": {
    "extra": [],
    "height": 512,
    "raw_info": "highly insanely detailed, masterpiece, top quality, best quality, highres, 4k, 8k, RAW photo, (very aesthetic, beautiful and aesthetic), 1girl, solo
Negative prompt: EasyNegative, (normal quality, low quality, worst quality:1.4), jpeg artifacts, (username, watermark, signature, time signature, timestamp, artist name, copyright name, copyright signature, copyright), (loli, child, infant, baby:1.3), (bad anatomy, extra digits, extra_fingers, wrong finger, inaccurate limb, bad hand, bad fingers), (African American, African:1.6), (tits, nipple:1.6), (pubic hair:1.6)
Steps: 20, Sampler: DPM++ 2M Karras, CFG scale: 7, Seed: 557152045, Size: 512x512, Model hash: 5307b134ff, Model: geminixMixRealistic_bV10, VAE hash: df3c506e51, VAE: kl-f8-anime2.ckpt, ADetailer model: lips_v1.pt, ADetailer prompt: "highly insanely detailed, masterpiece, top quality, best quality, highres, 4k, 8k, RAW photo, (very aesthetic, beautiful and aesthetic), (Vietnamese lips:1.2)", ADetailer confidence: 0.3, ADetailer dilate erode: 4, ADetailer mask blur: 4, ADetailer denoising strength: 0.4, ADetailer inpaint only masked: True, ADetailer inpaint padding: 32, ADetailer ControlNet model: Passthrough, ADetailer model 2nd: hand_yolov8n.pt, ADetailer prompt 2nd: "(perfect face, beautiful face, perfect lips, perfect mouth, perfect detailed leg, beautiful detailed leg, perfect accurate limb, perfect beautiful breasts, perfect body, perfect anatomy), (perfect hands, perfect hand anatomy, perfect finger, beautiful finger, perfect hands, good hands:1.2), beautiful and aesthetic, [PROMPT]", ADetailer confidence 2nd: 0.3, ADetailer dilate erode 2nd: 4, ADetailer mask blur 2nd: 4, ADetailer denoising strength 2nd: 0.4, ADetailer inpaint only masked 2nd: True, ADetailer inpaint padding 2nd: 32, ADetailer version: 24.4.2, sv_prompt: "highly insanely detailed, masterpiece, top quality, best quality, highres, 4k, 8k, RAW photo, (very aesthetic, beautiful and aesthetic), 1girl, solo", sv_negative: "EasyNegative, (normal quality, low quality, worst quality:1.4), jpeg artifacts, (username, watermark, signature, time signature, timestamp, artist name, copyright name, copyright signature, copyright), (loli, child, infant, baby:1.3), (bad anatomy, extra digits, extra_fingers, wrong finger, inaccurate limb, bad hand, bad fingers), (African American, African:1.6), (tits, nipple:1.6), (pubic hair:1.6)", Template: "highly insanely detailed, masterpiece, top quality, best quality, highres, 4k, 8k, RAW photo, (very aesthetic, beautiful and aesthetic), 1girl, solo", Negative Template: "EasyNegative, (normal quality, low quality, worst quality:1.4), jpeg artifacts, (username, watermark, signature, time signature, timestamp, artist name, copyright name, copyright signature, copyright), (loli, child, infant, baby:1.3), (bad anatomy, extra digits, extra_fingers, wrong finger, inaccurate limb, bad hand, bad fingers), (African American, African:1.6), (tits, nipple:1.6), (pubic hair:1.6)", Cutoff enabled: True, Cutoff targets: ["white", "black", "green", "red", "blue", "yellow", "pink", "purple", "bronze", "blonde", "silver", "magenta"], Cutoff padding: _</w>, Cutoff weight: 0.5, Cutoff disable_for_neg: True, Cutoff strong: False, Cutoff interpolation: lerp, TI hashes: "EasyNegative: c74b4e810b03", Version: f0.0.17v1.8.0rc-1.7.0",
    "width": 512,
  },
  "pnginfo": {
    "ADetailer ControlNet model": "Passthrough",
    "ADetailer confidence": 0.3,
    "ADetailer confidence 2nd": 0.3,
    "ADetailer denoising strength": 0.4,
    "ADetailer denoising strength 2nd": 0.4,
    "ADetailer dilate erode": 4,
    "ADetailer dilate erode 2nd": 4,
    "ADetailer inpaint only masked": "True",
    "ADetailer inpaint only masked 2nd": "True",
    "ADetailer inpaint padding": 32,
    "ADetailer inpaint padding 2nd": 32,
    "ADetailer mask blur": 4,
    "ADetailer mask blur 2nd": 4,
    "ADetailer model": "lips_v1.pt",
    "ADetailer model 2nd": "hand_yolov8n.pt",
    "ADetailer prompt": ""highly insanely detailed, masterpiece, top quality, best quality, highres, 4k, 8k, RAW photo, (very aesthetic, beautiful and aesthetic), (Vietnamese lips:1.2)"",
    "ADetailer prompt 2nd": ""(perfect face, beautiful face, perfect lips, perfect mouth, perfect detailed leg, beautiful detailed leg, perfect accurate limb, perfect beautiful breasts, perfect body, perfect anatomy), (perfect hands, perfect hand anatomy, perfect finger, beautiful finger, perfect hands, good hands:1.2), beautiful and aesthetic, [PROMPT]"",
    "ADetailer version": "24.4.2",
    "CFG scale": 7,
    "Cutoff disable_for_neg": "True",
    "Cutoff enabled": "True",
    "Cutoff interpolation": "lerp",
    "Cutoff padding": "_</w>",
    "Cutoff strong": "False",
    "Cutoff targets": "["white", "black", "green", "red", "blue", "yellow", "pink", "purple", "bronze", "blonde", "silver", "magenta"]",
    "Cutoff weight": 0.5,
    "Model": "geminixMixRealistic_bV10",
    "Model hash": "5307b134ff",
    "Negative Template": ""EasyNegative, (normal quality, low quality, worst quality:1.4), jpeg artifacts, (username, watermark, signature, time signature, timestamp, artist name, copyright name, copyright signature, copyright), (loli, child, infant, baby:1.3), (bad anatomy, extra digits, extra_fingers, wrong finger, inaccurate limb, bad hand, bad fingers), (African American, African:1.6), (tits, nipple:1.6), (pubic hair:1.6)"",
    "Sampler": "DPM++ 2M Karras",
    "Seed": 557152045,
    "Size": "512x512",
    "Steps": 20,
    "TI hashes": ""EasyNegative: c74b4e810b03"",
    "Template": ""highly insanely detailed, masterpiece, top quality, best quality, highres, 4k, 8k, RAW photo, (very aesthetic, beautiful and aesthetic), 1girl, solo"",
    "VAE": "kl-f8-anime2.ckpt",
    "VAE hash": "df3c506e51",
    "Version": "f0.0.17v1.8.0rc-1.7.0",
    "negative_prompt": "EasyNegative, (normal quality, low quality, worst quality:1.4), jpeg artifacts, (username, watermark, signature, time signature, timestamp, artist name, copyright name, copyright signature, copyright), (loli, child, infant, baby:1.3), (bad anatomy, extra digits, extra_fingers, wrong finger, inaccurate limb, bad hand, bad fingers), (African American, African:1.6), (tits, nipple:1.6), (pubic hair:1.6)",
    "prompt": "highly insanely detailed, masterpiece, top quality, best quality, highres, 4k, 8k, RAW photo, (very aesthetic, beautiful and aesthetic), 1girl, solo",
    "sv_negative": ""EasyNegative, (normal quality, low quality, worst quality:1.4), jpeg artifacts, (username, watermark, signature, time signature, timestamp, artist name, copyright name, copyright signature, copyright), (loli, child, infant, baby:1.3), (bad anatomy, extra digits, extra_fingers, wrong finger, inaccurate limb, bad hand, bad fingers), (African American, African:1.6), (tits, nipple:1.6), (pubic hair:1.6)"",
    "sv_prompt": ""highly insanely detailed, masterpiece, top quality, best quality, highres, 4k, 8k, RAW photo, (very aesthetic, beautiful and aesthetic), 1girl, solo"",
  },
}
`;

exports[`info.spec PNGINFO:cast_to_snake 1`] = `
{
  "metadata": {
    "extra": [],
    "height": 512,
    "raw_info": "highly insanely detailed, masterpiece, top quality, best quality, highres, 4k, 8k, RAW photo, (very aesthetic, beautiful and aesthetic), 1girl, solo
Negative prompt: EasyNegative, (normal quality, low quality, worst quality:1.4), jpeg artifacts, (username, watermark, signature, time signature, timestamp, artist name, copyright name, copyright signature, copyright), (loli, child, infant, baby:1.3), (bad anatomy, extra digits, extra_fingers, wrong finger, inaccurate limb, bad hand, bad fingers), (African American, African:1.6), (tits, nipple:1.6), (pubic hair:1.6)
Steps: 20, Sampler: DPM++ 2M Karras, CFG scale: 7, Seed: 557152045, Size: 512x512, Model hash: 5307b134ff, Model: geminixMixRealistic_bV10, VAE hash: df3c506e51, VAE: kl-f8-anime2.ckpt, ADetailer model: lips_v1.pt, ADetailer prompt: "highly insanely detailed, masterpiece, top quality, best quality, highres, 4k, 8k, RAW photo, (very aesthetic, beautiful and aesthetic), (Vietnamese lips:1.2)", ADetailer confidence: 0.3, ADetailer dilate erode: 4, ADetailer mask blur: 4, ADetailer denoising strength: 0.4, ADetailer inpaint only masked: True, ADetailer inpaint padding: 32, ADetailer ControlNet model: Passthrough, ADetailer model 2nd: hand_yolov8n.pt, ADetailer prompt 2nd: "(perfect face, beautiful face, perfect lips, perfect mouth, perfect detailed leg, beautiful detailed leg, perfect accurate limb, perfect beautiful breasts, perfect body, perfect anatomy), (perfect hands, perfect hand anatomy, perfect finger, beautiful finger, perfect hands, good hands:1.2), beautiful and aesthetic, [PROMPT]", ADetailer confidence 2nd: 0.3, ADetailer dilate erode 2nd: 4, ADetailer mask blur 2nd: 4, ADetailer denoising strength 2nd: 0.4, ADetailer inpaint only masked 2nd: True, ADetailer inpaint padding 2nd: 32, ADetailer version: 24.4.2, sv_prompt: "highly insanely detailed, masterpiece, top quality, best quality, highres, 4k, 8k, RAW photo, (very aesthetic, beautiful and aesthetic), 1girl, solo", sv_negative: "EasyNegative, (normal quality, low quality, worst quality:1.4), jpeg artifacts, (username, watermark, signature, time signature, timestamp, artist name, copyright name, copyright signature, copyright), (loli, child, infant, baby:1.3), (bad anatomy, extra digits, extra_fingers, wrong finger, inaccurate limb, bad hand, bad fingers), (African American, African:1.6), (tits, nipple:1.6), (pubic hair:1.6)", Template: "highly insanely detailed, masterpiece, top quality, best quality, highres, 4k, 8k, RAW photo, (very aesthetic, beautiful and aesthetic), 1girl, solo", Negative Template: "EasyNegative, (normal quality, low quality, worst quality:1.4), jpeg artifacts, (username, watermark, signature, time signature, timestamp, artist name, copyright name, copyright signature, copyright), (loli, child, infant, baby:1.3), (bad anatomy, extra digits, extra_fingers, wrong finger, inaccurate limb, bad hand, bad fingers), (African American, African:1.6), (tits, nipple:1.6), (pubic hair:1.6)", Cutoff enabled: True, Cutoff targets: ["white", "black", "green", "red", "blue", "yellow", "pink", "purple", "bronze", "blonde", "silver", "magenta"], Cutoff padding: _</w>, Cutoff weight: 0.5, Cutoff disable_for_neg: True, Cutoff strong: False, Cutoff interpolation: lerp, TI hashes: "EasyNegative: c74b4e810b03", Version: f0.0.17v1.8.0rc-1.7.0",
    "width": 512,
  },
  "pnginfo": {
    "adetailer_confidence": 0.3,
    "adetailer_confidence_2nd": 0.3,
    "adetailer_controlnet_model": "Passthrough",
    "adetailer_denoising_strength": 0.4,
    "adetailer_denoising_strength_2nd": 0.4,
    "adetailer_dilate_erode": 4,
    "adetailer_dilate_erode_2nd": 4,
    "adetailer_inpaint_only_masked": "True",
    "adetailer_inpaint_only_masked_2nd": "True",
    "adetailer_inpaint_padding": 32,
    "adetailer_inpaint_padding_2nd": 32,
    "adetailer_mask_blur": 4,
    "adetailer_mask_blur_2nd": 4,
    "adetailer_model": "lips_v1.pt",
    "adetailer_model_2nd": "hand_yolov8n.pt",
    "adetailer_prompt": ""highly insanely detailed, masterpiece, top quality, best quality, highres, 4k, 8k, RAW photo, (very aesthetic, beautiful and aesthetic), (Vietnamese lips:1.2)"",
    "adetailer_prompt_2nd": ""(perfect face, beautiful face, perfect lips, perfect mouth, perfect detailed leg, beautiful detailed leg, perfect accurate limb, perfect beautiful breasts, perfect body, perfect anatomy), (perfect hands, perfect hand anatomy, perfect finger, beautiful finger, perfect hands, good hands:1.2), beautiful and aesthetic, [PROMPT]"",
    "adetailer_version": "24.4.2",
    "cfg_scale": 7,
    "cutoff_disable_for_neg": "True",
    "cutoff_enabled": "True",
    "cutoff_interpolation": "lerp",
    "cutoff_padding": "_</w>",
    "cutoff_strong": "False",
    "cutoff_targets": "["white", "black", "green", "red", "blue", "yellow", "pink", "purple", "bronze", "blonde", "silver", "magenta"]",
    "cutoff_weight": 0.5,
    "model": "geminixMixRealistic_bV10",
    "model_hash": "5307b134ff",
    "negative_prompt": "EasyNegative, (normal quality, low quality, worst quality:1.4), jpeg artifacts, (username, watermark, signature, time signature, timestamp, artist name, copyright name, copyright signature, copyright), (loli, child, infant, baby:1.3), (bad anatomy, extra digits, extra_fingers, wrong finger, inaccurate limb, bad hand, bad fingers), (African American, African:1.6), (tits, nipple:1.6), (pubic hair:1.6)",
    "negative_template": ""EasyNegative, (normal quality, low quality, worst quality:1.4), jpeg artifacts, (username, watermark, signature, time signature, timestamp, artist name, copyright name, copyright signature, copyright), (loli, child, infant, baby:1.3), (bad anatomy, extra digits, extra_fingers, wrong finger, inaccurate limb, bad hand, bad fingers), (African American, African:1.6), (tits, nipple:1.6), (pubic hair:1.6)"",
    "prompt": "highly insanely detailed, masterpiece, top quality, best quality, highres, 4k, 8k, RAW photo, (very aesthetic, beautiful and aesthetic), 1girl, solo",
    "sampler": "DPM++ 2M Karras",
    "seed": 557152045,
    "size": "512x512",
    "steps": 20,
    "sv_negative": ""EasyNegative, (normal quality, low quality, worst quality:1.4), jpeg artifacts, (username, watermark, signature, time signature, timestamp, artist name, copyright name, copyright signature, copyright), (loli, child, infant, baby:1.3), (bad anatomy, extra digits, extra_fingers, wrong finger, inaccurate limb, bad hand, bad fingers), (African American, African:1.6), (tits, nipple:1.6), (pubic hair:1.6)"",
    "sv_prompt": ""highly insanely detailed, masterpiece, top quality, best quality, highres, 4k, 8k, RAW photo, (very aesthetic, beautiful and aesthetic), 1girl, solo"",
    "template": ""highly insanely detailed, masterpiece, top quality, best quality, highres, 4k, 8k, RAW photo, (very aesthetic, beautiful and aesthetic), 1girl, solo"",
    "ti_hashes": ""EasyNegative: c74b4e810b03"",
    "vae": "kl-f8-anime2.ckpt",
    "vae_hash": "df3c506e51",
    "version": "f0.0.17v1.8.0rc-1.7.0",
  },
}
`;
